mycroft/skills/core.py

```diff
@@ -721,9 +721,22 @@ def voc_match(self, utt, voc_filename, lang=None):
                 self.voc_match_cache[cache_key] = f.read().splitlines()
 
         # Check for match
-        if utt and any(i.strip() in utt
-                       for i in self.voc_match_cache[cache_key]):
-            return True
+        if utt:
+            words = utt.split()
+            for i in self.voc_match_cache[cache_key]:
+                term = i.strip()
+                if not term or term[0] is "#":  # ignore blanks and comments
```

Comment:
Let's do this when reading lines into the cache instead... Also we need to run the expand parenthesis on this as well